### PR TITLE
Bump nmos-cpp's zlib requirement to [>=1.2.11 <2] range

### PR DESCRIPTION
**Pull request by RubenRBS**

### CODE REVIEW

File: recipes/nmos-cpp/all/conanfile.py [hunk 1]
```diff
@@ -65,7 +67,7 @@ def requirements(self):
 
     def build_requirements(self):
         # nmos-cpp needs CMake 3.17 or higher but CCI doesn't allow version ranges
-        self.build_requires("cmake/3.24.2")
+        self.build_requires("cmake/[>=3.17 <4]")
```
Comment (SpaceIm) [inline recipes/nmos-cpp/all/conanfile.py:70]:
```suggestion
        self.tool_requires("cmake/[>=3.17 <4]")
```

File: recipes/nmos-cpp/all/conanfile.py [hunk 2]
```diff
@@ -65,7 +67,7 @@ def requirements(self):
 
     def build_requirements(self):
         # nmos-cpp needs CMake 3.17 or higher but CCI doesn't allow version ranges
```
Comment (SpaceIm) [inline recipes/nmos-cpp/all/conanfile.py:69]:
```suggestion
```

This comment is not relevant anymore if you use version range.

**Comment by SpaceIm** [comment]
There is also this code to remove:

```python
    def package_id(self):
        self.info.requires["boost"].minor_mode()
        self.info.requires["nlohmann_json"].patch_mode()
```

File: recipes/nmos-cpp/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/nmos-cpp/all/conanfile.py]: Done.

File: recipes/nmos-cpp/all/conanfile.py (hunk 2 above)
Comment (uilianries) [reply recipes/nmos-cpp/all/conanfile.py]: Done.

**Comment by uilianries** [comment]
@SpaceIm Thank you for your review, all suggestions are applied now.

@RubenRBS I started a Tapaholes to generate `avahi/0.8`

**Review by jwillikers — APPROVED** [review]

**Comment by uilianries** [comment]
I just rebuilt `avahi/0.8` internally and now restarted this PR.

**Comment by RubenRBS** [comment]
Relaunched again, this time due to libselinux

**Review by czoido — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by uilianries** [comment]
Avahi missing again?! I'll check it further.

**Comment by uilianries** [comment]
I created another PR (https://github.com/conan-io/conan-center-index/pull/20706) that should fix avahi. That's why we still have it missing.

**Review by uilianries — APPROVED** [review]

**Pull request merged**